File: src/ru/henridellal/dialer/SpeedDial.java
package ru.henridellal.dialer;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class SpeedDial {
	
	private static final String KEY = "speed_dial_key_";
	
	public static final String getNumber(Context context, String order) {
		return PreferenceManager.getDefaultSharedPreferences(context)
				.getString(KEY + order, "");
	}
	
	public static final void setNumber(Context context, String order, String number) {
		if (number == null || number.length() == 0 || null == order) {
			return;
		}
		
		SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(context).edit();
		editor.putString(KEY + order, number).commit();
	}
	
	public static final void clearSlot(Context context, String order) {
		SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(context).edit();
		editor.putString(KEY + order, "").commit();
	}
	
}

File: app/src/main/java/ru/henridellal/dialer/SpeedDial.kt
package ru.henridellal.dialer

import android.content.Context
import android.preference.PreferenceManager

object SpeedDial {
    private const val KEY = "speed_dial_key_"
    @JvmStatic
	fun getNumber(context: Context?, order: String): String? {
        return PreferenceManager.getDefaultSharedPreferences(context)
                .getString(KEY + order, "")
    }

    @JvmStatic
	fun setNumber(context: Context?, order: String?, number: String?) {
        if (number == null || number.isEmpty() || null == order) {
            return
        }
        val editor = PreferenceManager.getDefaultSharedPreferences(context).edit()
        editor.putString(KEY + order, number).apply()
    }

    @JvmStatic
	fun clearSlot(context: Context?, order: String) {
        val editor = PreferenceManager.getDefaultSharedPreferences(context).edit()
        editor.putString(KEY + order, "").apply()
    }
}